source/backends/debian/debpkgindex.d

```diff
@@ -195,7 +200,7 @@ public:
             if (!name)
                 continue;
 
-            auto pkg = new DebPackage (name, ver, arch);
+            auto pkg = newPackage (name, ver, arch);
```

Comment:
The nicest way I can think of would be making use of D's awesome templating system.
That would allow you to write one function for `DebPackage` and Ubuntu packages which gets compiled for the right type at build-time.

E.g.:
```D
private T[] loadPackages(T) (string suite, string section, string arch)
    if (is(T == DebPackage) || is(T == UbuntuPackage))
{
    auto pkg = new T (name, ver, arch);
    static if (is(T == UbuntuPackage)) {
        // Ubuntu stuff
    }
}
```

The current solution isn't very unclean though, so I would be okay with that too (the templating stuff is probably the cleanest way though).